iceoryx_examples/ara_com/ara/com/event_publisher.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_ARA_COM_EVENT_PUBLISHER_HPP
+#define IOX_EXAMPLES_ARA_COM_EVENT_PUBLISHER_HPP
+
+#include "iceoryx_hoofs/cxx/optional.hpp"
+#include "iceoryx_posh/popo/publisher.hpp"
+
+#include "ara/types.hpp"
+
+#include <memory>
+#include <utility>
+
+class MinimalSkeleton;
+
+namespace ara
+{
+namespace com
+{
+template <typename T>
+class EventPublisher
+{
+  public:
+    using SampleType = T;
+
+    // Deleted because of Allocate() capturing 'this'
+    EventPublisher(const EventPublisher&) = delete;
+    EventPublisher(EventPublisher&&) = delete;
+    EventPublisher& operator=(const EventPublisher&) = delete;
+    EventPublisher& operator=(EventPublisher&&) = delete;
+
+    EventPublisher(const core::String& service, const core::String& instance, const core::String& event) noexcept
+        : m_publisher({service, instance, event}, {1U, "", false})
+    {
+    }
+
+    void Send(const SampleType& userSample) noexcept
```

Comment:
AFAIK the AUTOSAR discourages the use of exceptions, but does not show this in the function signatures. For an iceoryx implemention I would suggest to use `noexcept` everywhere, except the part of which uses `std::future`'s.

Reply:
@mossmaurice I think we have the discussion to stop using noexcept and in fact remove it everywhere where it does not apply, i.e. we cannot guarantee it due to e.g. calling a hook/callback function we cannot control. I think it is fine to not define the functions with `noexcept` if Autosar does not require it but I am fine either way.

I always preferred `noexcept` only(!) if it really cannot throw. Otherwise it does more harm than good.